### PR TITLE
Allow customization of DXC executable

### LINKED ISSUE #17148: DXC should be customizable via Settings Registry
**Is your feature request related to a problem? Please describe.**
Being able to customize DXC is important as an option to workaround bugs like this one:
https://github.com/o3de/o3de-extras/issues/625.  
Alternatively it would make it easier to try experimental versions of DXC, etc.

O3DE already supports customization of **AZSLc** via the following Settings Registry Key:  
`/O3DE/Atom/AzslCompilerOverridePath`.

**Describe the solution you'd like**
Allow customization of DXC via the following Settings Registry Key:  
`/O3DE/Atom/DxcOverridePath`.

**Describe alternatives you've considered**
N/A

**Additional context**
See https://github.com/o3de/o3de-extras/issues/625.

### PR DESCRIPTION
**Pull request by galibzon**
## What does this PR do?

Resolves #17148 (DXC should be customizable via Settings Registry) Which enables working around this bug:
https://github.com/o3de/o3de-extras/issues/625

Developers can now pick a custom version of the DXC executable via the Settings Registry Key:
`/O3DE/Atom/DxcOverridePath`.

Also, introduced the macro `PRE_HLSL_2021`, along with an explanation in the following file:
`/o3de/Gems/Atom/Feature/Common/Assets/ShaderLib/PRE_HLSL_2021.md `.  
TL;DR: If a game project uses a DXC executable that doesn't conform to the HLSL 2021 standard, then they should globally define the macro PRE_HLSL_2021  when compiling shaders.

The O3DE default version of DXC is 1.7.2308 which conforms to HLSL 2021. The problem is that some shaders get compiled to versions of SPIRV not supported by Adreno GPU drivers. This commit gives the option to use a different DXC and avoid this issue.

Removed redundant LinearToSRGB conversion functions.

## How was this PR tested?

Validated the OpenXRTest project running on Meta Quest Pro device.

### CODE REVIEW

**Review by antonmic — APPROVED** [review]

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli [hunk 1]
```diff
@@ -8,41 +8,9 @@
 
 #pragma once
 
-/*
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli:11]: These functions do get used by non-Atom gems. Please add them back. Thanks

Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli:11]: Currently there is real keyword version of Srgb_To_LinearSrgb and LinearSrgb_To_Srgb. Maybe follow the same pattern here and add a **float** one and a **half** one. That way unnecessary up/down casts can be avoided from different shaders on mobile plat.

Comment (galibzon) [reply Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli:11]: Isn't real3 supposed to take care of those issues?

Comment (moudgils) [reply Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli:11]: Yeah but not al the shaders are converted. So if this function is called from a shader like the UI one then on mobile you are up/down casting for just one ALU instruction and is not worth it. Having two overloaded functions avoid this issue as the compiler will pick the correct one based on type.

**Review by AMZN-alexpete — APPROVED** [review]

**Review by jhmueller-huawei — APPROVED** [review]

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli (hunk 1 above)
Comment (galibzon) [reply Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PostProcessing/PostProcessUtil.azsli:11]: Got it. Added the `float3 LinearSrgb_To_SrgbF(float3 color)` and `float3 Srgb_To_LinearSrgbF(float3 color)` and
`ImGuiAtom.azsl` and `LyShineUI.azsl` now call those functions.

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/ColorManagement/GeneratedTransforms/Srgb_To_LinearSrgb.azsli [hunk 2]
```diff
@@ -11,5 +11,25 @@
 real3 Srgb_To_LinearSrgb(real3 color)
 {
     // Copied from CryFx's ToAccurateLinear()
+#ifdef PRE_HLSL_2021
+    // See /o3de/Gems/Atom/Feature/Common/Assets/ShaderLib/PRE_HLSL_2021.md for details.
+    return (color.xyz < 0.04045) ? color.xyz / 12.92h : pow((color.xyz + real3(0.055, 0.055, 0.055)) / real3(1.055, 1.055, 1.055), 2.4);
+#else
     return select((color.xyz < 0.04045), color.xyz / 12.92h, pow((color.xyz + real3(0.055, 0.055, 0.055)) / real3(1.055, 1.055, 1.055), 2.4));
+#endif
+}
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/ColorManagement/GeneratedTransforms/Srgb_To_LinearSrgb.azsli:20]: What I meant with my previous comment is to have the same method name but different type. So you will need the following two versions with same function name. 

half3 Srgb_To_LinearSrgb(half3 color)
and
float3 Srgb_To_LinearSrgb(float3 color)
No reals needed here. And the same applies to LinearSrgb_To_Srgb

Comment (galibzon) [reply Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/ColorManagement/GeneratedTransforms/Srgb_To_LinearSrgb.azsli:20]: Oh, sorry: Was thrown off by this comment in the code:
```
/* AZSL doesn't support function overload for the moment.
half3 LinearToSRGB(half3 col)
```
Fixed it now.

**Review by moudgils — APPROVED** [review]
